analyzer/src/main/kotlin/PackageManager.kt

```diff
@@ -272,7 +273,7 @@ abstract class PackageManager(
 
         afterResolution(definitionFiles)
 
-        return createPackageManagerResult(result)
+        return createPackageManagerResult(result).addDependencyGraphIfMissing()
```

Comment:
It would be good to have a unit test to expose the bug; we could then easier check different approaches to solve it.

Reply:
I've spent some time to add a (unit) test, but it's not easy because the invariants go across components (PackageManager, CompatibilityDependencyNavigator.create()). If the plan is to remove the Tree representation / Compatibility navigator, would it be ok to live without a test for this bug?

Reply:
My first idea was to construct an `OrtResult` with a single project that has an empty set of scope names and no dependency graphs. When then the `DependencyNavigator` is queried, an exception is currently thrown.

But I see now that this test already depends on a solution. With your approach, such a result would not be created by the `PackageManager`. - But wouldn't this be another argument for not solving the problem in `PackageManager`? `OrtResult` should not fail to return a navigator, no matter what projects it contains?

Reply:
> But I see now that this test already depends on a solution. With your approach, such a result would not be created by the `PackageManager`. - But wouldn't this be another argument for not solving the problem in `PackageManager`? `OrtResult` should not fail to return a navigator, no matter what projects it contains?

So what solution exactly do you have in mind?

Reply:
Changing the `CompatibilityDependencyNavigator` to handle this corner case; namely to modify the `usesTree()` function as I have proposed. Maybe it should then be given a better name.